Source/WebKit/UIProcess/WebAuthentication/Cocoa/WebAuthenticatorCoordinatorProxy.mm

```diff
@@ -237,7 +245,14 @@ - (void)authorizationController:(ASAuthorizationController *)controller didCompl
         }
     }
     if ([platformAllowedCredentials count] || ![crossPlatformAllowedCredentials count]) {
-        auto request = adoptNS([[allocASAuthorizationPlatformPublicKeyCredentialProviderInstance() initWithRelyingPartyIdentifier:options.rpId] createCredentialAssertionRequestWithChallenge:toNSData(options.challenge).get()]);
+        RetainPtr<ASAuthorizationPlatformPublicKeyCredentialAssertionRequest> request;
+        RetainPtr<ASAuthorizationPlatformPublicKeyCredentialProvider> provider = adoptNS([allocASAuthorizationPlatformPublicKeyCredentialProviderInstance() initWithRelyingPartyIdentifier:options.rpId]);
+        if ([provider.get() respondsToSelector:@selector(createCredentialAssertionRequestWithClientData:)]) {
+            RetainPtr<ASPublicKeyCredentialClientData> clientData = adoptNS([allocASPublicKeyCredentialClientDataInstance() initWithChallenge:toNSData(options.challenge).get() origin:origin]);
+            request = adoptNS([provider.get() createCredentialAssertionRequestWithClientData:clientData.get()]);
+        } else
+            request = adoptNS([provider.get() createCredentialAssertionRequestWithChallenge:toNSData(options.challenge).get()]);
```

Comment:
nit: You don't need to use .get() whenever something is on the receiver side of a obj-c style call. i.e. [provider.get() ...] can be [provider ...].